### PR TITLE
[aievec] remove aie1 ops from aievec dialect

### PR DESCRIPTION
**Pull request by jamestcl-amd**
This is a follow-up PR to https://github.com/Xilinx/mlir-aie/pull/1622/. This PR includes the following changes:

- Removes the AIE1 ops `add/sub/mac/mul` from the existing AIEVEC.
- Includes header `AIEVecAIE1Dialect` and `AIEVecAIE1Ops` wherever needed.
- Updates conversion passes to create AIE1 ops from `AIEVecAIE1Dialect`.
- Migrates all existing AIE1 ops aievec tests to use the new ops in `AIEVecAIE1Dialect`.
- The `AIEVecAIE1Dialect` is not exposed to the Python interface and hence the `mul_mul()` test in `python/aievec.py` is modified to use `aievec.mul_elem` in AIEVEC. 

Todo:
- From reviewing the conversion passes, the `AIEVecAIE1Dialect` also needs `ups/srs/upd/concat/ext/select/pack` ops. This can be done in another follow-up PR.

### CODE REVIEW

File: test/python/aievec.py [hunk 1]
```diff
@@ -46,39 +46,23 @@ def demo_fun1():
...
-        v1 = aievec.concat([v0, v0])
-        for i in scf.for_(0, 2048, 8):
-            v2 = aievec.upd(T.vector(8, T.f32()), A, [i])
-            v3 = aievec.upd(T.vector(8, T.f32()), B, [i])
-            v4 = aievec.mul(
-                T.vector(8, T.f32()),
+        for i in scf.for_(0, 2048, 32):
+            v0 = aievec.upd(T.vector(32, T.i16()), A, [i])
+            v1 = aievec.upd(T.vector(32, T.i16()), B, [i])
+            v2 = aievec.mul_elem(
```
Comment (david-vc) [inline test/python/aievec.py:57]: The original test was targeting AIE1.
Are the aievec.aie1 ops no longer accessible through the python interface?

**Review by david-vc — APPROVED** [review]
LGTM, but perhaps the PR description should clarify that support for aievec.aie1 in the python interface is not done.

**Review by jsetoain — APPROVED** [review]
Thanks, James!

**Comment by jamestcl-amd** [comment]
Will clean up all the leftover comments and after https://github.com/Xilinx/mlir-aie/pull/1636 is merged.

File: test/python/aievec.py (hunk 1 above)
Comment (jamestcl-amd) [reply test/python/aievec.py:57]: Yes, right. The `AIEVecAIE1Dialect` is not exposed to the Python interface.

**Pull request merged**